### PR TITLE
add awssm support

### PR DESCRIPTION
**Pull request by moolen**
TODO:
- [x] add more tests to cover the missing bits (see codecov comments)

### CODE REVIEW

File: pkg/controllers/externalsecret/externalsecret_controller.go [hunk 1]
```diff
@@ -127,6 +134,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) shouldProcessStore(store esv1alpha1.GenericStore) bool {
+	if store.GetSpec().Controller == "" || store.GetSpec().Controller != r.ControllerClass {
```
Comment (mcavoyk) [inline pkg/controllers/externalsecret/externalsecret_controller.go:138]:
```suggestion
	if store.GetSpec().Controller == "" || store.GetSpec().Controller == r.ControllerClass {
```

I believe the logic is that the controller manages the store if the controller class is unspecified or that it matches the controllerClass

Comment (moolen) [reply pkg/controllers/externalsecret/externalsecret_controller.go]: yes, thanks! That's one of the things that aren't properly tested yet :)

**Comment by moolen** [comment]
I've added a bunch of tests, feel free to take a look if you have  some time

**Review by mcavoyk — APPROVED** [review]
LGTM

**Pull request merged**